### PR TITLE
Add paravirtual mode support, nodeSelector, deployment Strategy for different versions of CPI.

### PR DESCRIPTION
**Pull request by HanFa**
## What this PR does / why we need it

This PR 

[x] enables support of para-virtual mode for older versions of CPI, to support 1.21 and 1.22 TKr.
[x] adds node selector and update strategy for 1.21.3, 1.22.6, 1.23.0 CPI package, with additional unit tests 

## Which issue(s) this PR fixes

Fixes: #

## Describe testing done for PR

CPI package unit test

### CODE REVIEW

File: addons/packages/vsphere-cpi/1.23.0/test/vsphere_cpi_test.go [hunk 1]
```diff
@@ -682,6 +757,15 @@ func parseDaemonSet(output string) appsv1.DaemonSet {
 	return daemonSet
 }
 
+func parseDeployment(output string) appsv1.Deployment {
+	deploymentDocIndex := 8
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0/test/vsphere_cpi_test.go:761]: This gonna bite us in the future.. anyway

Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0/test/vsphere_cpi_test.go:761]: hahaha

**Comment by lubronzhan** [comment]
/lgtm
@shyaamsn please take a look

**Review by lubronzhan — APPROVED** [review]

**Pull request merged**